### DISCUSSION TITLE
window.lua too long w/o yielding

### DISCUSSION THREAD

**Discussion by Yabastar** [cc-tweaked/CC-Tweaked#1578] [Q&A]
### Minecraft Version

1.18.x

### Version

1.101.3

### Details

![image](https://github.com/cc-tweaked/CC-Tweaked/assets/138197982/c4d2890d-44fa-47f9-b5b8-1ca6315ab086)
https://pastebin.com/CLDisqHm

**Comment by SquidDev** [comment]
CC: Tweaked will terminate code if it runs for too long, in order to prevent it wasting server resources.

In your case, you're calling `peripheral.wrap` $4 \times 201^3 \approx 32 \text{ million}$ times, which is far too many! Adding some `sleep()`s (for instance, after 14) will cause the code to yield and prevent this error.

However, in this particular case, I'm not entirely sure if this code does what you want it to. I don't think there is a trapdoor peripheral, so don't think this will ever do anything.
